docs/usage.md

```diff
@@ -25,20 +25,36 @@ TODO
 
 ## Generate models from AsyncAPI documents
 
+When providing an AsyncAPI document, the library iterates the entire document and generate models for all defined messages. If any other kind of iteration is wanted, feel free to create a [feature request](https://github.com/asyncapi/modelina/issues/new?assignees=&labels=enhancement&template=enhancement.md).
+
 There are two ways to generate models for an AsyncAPI document.
 
 - [Generate from a parsed AsyncAPI document](../examples/asyncapi-from-parser)
 - [Generate from a pure JS object](../examples/asyncapi-from-object)
 
 The library expects the `asyncapi` property for the document to be set in order to understand the input format.
 
+The message payloads, since it is a JSON Schema variant, is [interpreted as a such](./interpretation_of_JSON_Schema.md).
+
 ## Generate models from JSON Schema draft 7 documents
 
 There are one way to generate models from a JSON Schema draft 7 document.
 
 - [Generate from a pure JS object](../examples/json-schema-draft7-from-object)
 
-The library expects the `$schema` property for the document to be set in order to understand the input format. By default, if no other inputs are detected, it defaults to `JSON Schema draft 7`.
+The library expects the `$schema` property for the document to be set in order to understand the input format. By default, if no other inputs are detected, it defaults to `JSON Schema draft 7`. The process of interpreting a JSON Schema to a model can be read [here](./interpretation_of_JSON_Schema.md).
+
+## Generate models from Swagger 2.0 documents
+When providing an AsyncAPI document, Modelina iterates the entire document and generate models for all defined body parameters and responses. If any other kind of iteration is wanted, feel free to create a [feature request](https://github.com/asyncapi/modelina/issues/new?assignees=&labels=enhancement&template=enhancement.md).
+
+There are one way to generate models from a Swagger 2.0 document
+
+- [Generate from a pure JS object](../examples/swagger2.0-from-object)
+
+The Swagger input processor expects that the property `swagger` is defined in order to know it should be processed.
+
+The response payload and parameters, since it is a JSON Schema variant, is [interpreted as a such](./interpretation_of_JSON_Schema.md).
```

Comment:
```suggestion
The response payload and `body` parameters, since it is a JSON Schema variant, is [interpreted as a such](./interpretation_of_JSON_Schema.md).
```